cellprofiler/modules/tram.py

```diff
@@ -0,0 +1,673 @@
+            These are measurements for the selected tracked objects which
+            will be used in the TrAM computation. At least one must be selected.
+            Note there may be a delay of a few seconds between the selection of
+            <i>Tracked objects</i> and the update of this selection component.""")
+
+        # Treat X-Y value pairs as Euclidian in the TrAM measure?
+        self.wants_XY_Euclidian = cps.Binary(
+            'Euclidian XY metric?', True, doc="""
+            If selected (the default) then measurements that are available
+            as X-Y pairs (e.g. location) will be have a Euclidian (isotropic)
+            metric applied in TrAM. Note that the X-Y-Z extension of this feature
+            is not currently available.
+            """)
+
+        # number of spline knots
+        self.num_knots = cps.Integer(
+            "Number of spline knots", 4, minval=MIN_NUM_KNOTS, doc="""
+            Number of spline knots</i>The number of knots (indpendent values) used
+            when computing smoothing splines. This should be around 1/5th the number
+            of frames for reasonably oversampled time lapse sequences, and must be 3
+            or greater. It is approximately the maximum number of wiggles expected in
+            well-tracked trajectories
+            """)
+
+        # TrAM exponent
+        self.p = cps.Float(
+            "TrAM exponent", 0.5, minval=0.01, maxval=1, doc="""
+            This number is between 0.01 and 1 (default 0.5), and specifies how
+            strongly simultaneous sudden changes in multiple features synergize in
+            the TrAM metric. A lower value signifies higher synergy (at the risk of
+            missing tracking failures that are reflected in only some of the features).
+            """)
+
+    def settings(self):
+        return [self.object_name, self.tram_measurements, self.wants_XY_Euclidian, self.num_knots, self.p]
+
+    def validate_module(self, pipeline):
+        '''Make sure that the user has selected at least one measurement for TrAM and that there are tracking data.'''
+        if len(self.get_selected_tram_measurements()) == 0:
+            raise cps.ValidationError(
+                    "Please select at least one TrAM measurement for tracking of %s" % self.object_name.get_value(),
+                    self.tram_measurements)
+
+        # check on available tracking columns for the selected object
+        obj_name = self.object_name.get_value()
+        mc = pipeline.get_measurement_columns()
+        num_tracking_cols = len([entry for entry in mc if entry[0] == obj_name and entry[1].startswith(trackobjects.F_PREFIX)])
+        if num_tracking_cols == 0:
+            msg = "No %s data available for %s. Please select an object with tracking data."\
+                  % (trackobjects.F_PREFIX, obj_name)
+            raise cps.ValidationError(msg, self.object_name)
+
+    def visible_settings(self):
```

Comment:
You can omit implementing the `visible_settings` method when it returns `self.settings()`.